### PR TITLE
[GP-4055] IconLinkButton

### PR DESCRIPTION
**Pull request by Farzan-Hashmi**
## IconLinkButton

tldr: `LinkButton` + `IconButton` -> `IconLinkButton`. 

We want a component that looks like an `IconButton` but for loading links. Up until this point, we have passing Icons into `LinkButton`s, but the resulting styling isn't correct (aka doesn't  have the same shape as `IconButton`). `IconButton` with an `onClick()` redirect isn't really good either since we can't wrap it with `Link` in NextJS to get prefetching 
benefits.

jira: https://cambly.atlassian.net/jira/software/c/projects/GP/boards/44?selectedIssue=GP-4054

## Next task(s):
wrap this with `Link` like `NextLinkButton`. also create a wrapper for this in express

### CODE REVIEW

File: packages/syntax-core/src/IconLinkButton/IconLinkButton.module.css [hunk 1]
```diff
@@ -0,0 +1,36 @@
+@import url("../IconButton/IconButton.module.css");
```
Comment (Farzan-Hashmi) [inline packages/syntax-core/src/IconLinkButton/IconLinkButton.module.css:1]: not rlly sure if this a css best-practice? I suppose I could alternatively import `IconButton.module.css` from inside `IconLinkButton.tsx` and use the styles there.

Comment (somethiiing) [reply packages/syntax-core/src/IconLinkButton/IconLinkButton.module.css:1]: i think you can just import the `IconButton` css file into IconLinkButton! that would make it so if any styles from IconButton change, it will change here too!

**Review by nathan-ong — APPROVED** [review]
Looks good to me! I'd probably wait for @somethiiing to make a pass

File: packages/syntax-core/src/IconLinkButton/IconLinkButton.module.css [hunk 2]
```diff
@@ -0,0 +1,5 @@
+.disabled {
```
[the remark this answers is not in the document]
Comment (Farzan-Hashmi) [reply packages/syntax-core/src/IconLinkButton/IconLinkButton.module.css:1]: I think we need a `pointer-events: none;` property in the disabled style

[the remark this answers is not in the document]
Comment (somethiiing) [reply packages/syntax-core/src/IconLinkButton/IconLinkButton.module.css:1]: we also need that on `pointer-events: none` on button too, it was just left out for some reason. feel free to just add that and use those instead

**Review by somethiiing — APPROVED** [review]

**Comment by somethiiing** [comment]
great job on this!
